src/aspire/source/xform.py

```diff
@@ -283,11 +328,12 @@ def _apply_xform(xform, im, indices, adjoint=False):
     A simple global function (i.e. not a method) that is capable of being cached by joblib.Memory object's `cache`
     method.
     """
+    xform_name = str(xform).split()[0].split('.')[-1]
```

Comment:
What's happening here? A comment would be good.

Reply:
This statement tries to extract the short xfrom name for the long string.  Comments added.

Reply:
Shouldn't that be a method of `Xform`?

Reply:
__name__ is not defined for all the xform classes. This statement here is to extract a short name from a long string such as 
<aspire.source.xform.Downsample object at 0x2b112a581f98>. I come go ahead to add the __name__ if you would like to do it in this PR.

Reply:
Isn't that just `xform.__class__.__name__`? That would avoid all the string manipulation stuff. All the same it would be good for this to be a method of `Xform` (something like `Xform.short_name`).

Reply:
afaict, you are parsing the default `repr` since `__repr__` is not defined. You might pack Joakim's suggestion into `__str__`, then you can say something like:

`xform_name = str(xform_instance)`

The `short_name` property is also pretty clear as to what that would mean in use.

Overloading the special variable `__name__` might lead to a bad time if it is not consistent with how Python expects it to be used.

Reply:
@junchaoxia .. when you want an object as a string for informational purposes, just use the object in a string context (i.e. `f"This is {obj}"`) and let the object worry about what it decides to print (open a new issue if it doesn't print what you would like).

Reply:
Thanks. Create a new issue #232 for implementing `__repr__`.

Reply:
Modified the issue #232 to implement `__str__`.